### PR TITLE
support for mobile layout

### PR DESCRIPTION
**Pull request by sotiwin**
### Summary 🎯

I added support for mobile layout: on mobile the table of content is displayed in the left sidebar: 

image

One thing is missing, but I couldn't figure it out. After clicking on something the sidebar remains opened. 

Additionally there are a few minor changes:
- to allow development in browser
- there was a react error on the console: each child in a list should have unique key

### CODE REVIEW

File: src/widgets/index.tsx [hunk 1]
```diff
@@ -7,10 +7,18 @@ import {
 } from '@remnote/plugin-sdk';
 import '../style.css';
 import '../App.css';
+import { isMobileOs } from '../lib/utils';
 
 async function onActivate(plugin: ReactRNPlugin) {
   // Register a sidebar widget.
-  await plugin.app.registerWidget('table_of_contents', WidgetLocation.RightSidebar, {
+  let os = await plugin.app.getOperatingSystem();
```
Comment (toxoxi) [inline src/widgets/index.tsx:14]: [IMO] using const would be better as long as no override after declaration

File: src/widgets/index.tsx [hunk 2]
```diff
@@ -7,10 +7,18 @@ import {
...
+
+  let location = WidgetLocation.RightSidebar;
+  if (isMobileOs(os)) {
+    location = WidgetLocation.LeftSidebar
+  }
```
Comment (toxoxi) [inline src/widgets/index.tsx:19]: [IMO] As same as the above, using const would be more robust.
```suggestion
  const location = isMobileOs(os) ? WidgetLocation.LeftSidebar : WidgetLocation.RightSidebar;
```

File: webpack.config.js [hunk 3]
```diff
@@ -111,6 +111,7 @@ if (isProd) {
     watchFiles: ['src/*'],
     headers: {
       'Access-Control-Allow-Origin': '*',
+      "Access-Control-Allow-Headers": "baggage, sentry-trace"
```
Comment (toxoxi) [inline webpack.config.js:114]: For what reason do we need? 👀

File: package.json [hunk 4]
```diff
@@ -37,7 +37,7 @@
     "ts-node": "^10.9.1",
     "typescript": "^4.7.4",
     "webpack": "^5.73.0",
-    "webpack-cli": "^4.10.0",
-    "webpack-dev-server": "^4.9.3"
+    "webpack-cli": "^5.1.4",
+    "webpack-dev-server": "^5.0.4"
```
Comment (toxoxi) [inline package.json:41]: Thanks for upgrading 👍

File: public/manifest.json [hunk 5]
```diff
@@ -1,21 +1,22 @@
...
   "requestNative": false,
   "requiredScopes": [
     {
       "type": "All",
-      "level": "ReadCreateModify"
+      "level": "Read"
     }
-  ]
+  ],
+  "unlisted": true
```
Comment (toxoxi) [inline public/manifest.json:21]: Would some changes be reverted before merge? (author/id/repoUrl/unlisted)

**Review by toxoxi — COMMENTED** [review]
I'm sorry for late review. I left some comments.

**Comment by toxoxi** [comment]
@sotiwin 
> What do you think of using the left sidebar? Maybe there are other options. For example put an icon in the top of the page, which would open table of contents in a floating window.

I guess the current approach is good because:
- it delivers value with fewer changes
- it naturally fits with the native feature (sidebar) of RemNote

However, a floating window could be great if we can develop it well. Feel free to try it if you're interested :)

File: webpack.config.js (hunk 3 above)
Comment (sotiwin) [reply webpack.config.js:114]: It was needed for dev server to work when using with remnote opened in a browser. See in the remnote template: https://github.com/remnoteio/remnote-plugin-template-react/commit/a9c1590787e6a4ea0bb5fb4610da4547edea9866

**Comment by sotiwin** [comment]
> @sotiwin
> 
> > What do you think of using the left sidebar? Maybe there are other options. For example put an icon in the top of the page, which would open table of contents in a floating window.
> 
> I guess the current approach is good because:
> 
> * it delivers value with fewer changes
> * it naturally fits with the native feature (sidebar) of RemNote
> 
> However, a floating window could be great if we can develop it well. Feel free to try it if you're interested :)

Okay, maybe in the future :)
For now I updated the merge request and reverted the manifest changes.

File: webpack.config.js (hunk 3 above)
Comment (toxoxi) [reply webpack.config.js:114]: I got it! I suppose remnote uses Sentry to track bugs and those headers are required to do it.

**Comment by toxoxi** [comment]
hmm, I couldn't check its behavior because of this error.
image

However, the screenshot looks good so I approve this. Thank you your contribution!

**Review by toxoxi — APPROVED** [review]

**Comment by toxoxi** [comment]
I'll merge it and upload to plugin store

**Comment by toxoxi** [comment]
> hmm, I couldn't check its behavior because of this error.

I had totally forgotten the tricky nature that it doesn't work with browser but with the native app.

Great work!

image

**Pull request merged**